pkg/commands/overrides_exporter.go

```diff
@@ -27,19 +27,27 @@ type OverridesExporterCommand struct {
 	presetsFilePath   string
 	refreshInterval   time.Duration
 
-	registry      *prometheus.Registry
-	overrideGauge *prometheus.GaugeVec
+	registry     *prometheus.Registry
+	presetsGauge *prometheus.GaugeVec
+
+	lastLimitsMtx sync.Mutex
+	lastLimits    map[string]*validation.Limits
 }
 
 func NewOverridesExporterCommand() *OverridesExporterCommand {
 	registry := prometheus.NewRegistry()
-	return &OverridesExporterCommand{
+	oc := &OverridesExporterCommand{
 		registry: registry,
-		overrideGauge: promauto.With(registry).NewGaugeVec(prometheus.GaugeOpts{
-			Name: "cortex_overrides",
-			Help: "Various different limits.",
-		}, []string{"limit_type", "type", "user"}),
+		presetsGauge: promauto.With(registry).NewGaugeVec(prometheus.GaugeOpts{
+			Name: "cortex_overrides_presets",
+			Help: "Preset limits.",
+		}, []string{"limit_type", "name"}),
+		lastLimits: map[string]*validation.Limits{},
 	}
+
+	registry.Register(validation.NewOverridesExporter(oc))
```

Comment:
We need to wrap registry with `type="tenant"` label, and register OverridesExporter into it.

Reply:
I removed that. I moved presets to their own metrics which I think is cleaner imo.

Reply:
Oh, sorry, I see that presets now use different metric name.